pgx: Migrate easily convertible cases to v5

spilchen opened the pull request:
Currently, our codebase uses a mix of jackc/pgx versions, with some areas relying on v4 and others on v5. The goal is to fully migrate to v5. This commit updates all the straightforward cases to use v5.

The changes were made using a sed script for consistency. Some v4 usages remain, as they require additional modifications to migrate to v5. These will be addressed in a follow-up commit. This commit focuses solely on mechanical changes to keep the review process simple.

Epic: none
Release note: none

Informs #137595

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139611)
<!-- Reviewable:end -->

spilchen committed:
pgx: Migrate easily convertible cases to v5

Currently, our codebase uses a mix of jackc/pgx versions, with some
areas relying on v4 and others on v5. The goal is to fully migrate to
v5. This commit updates all the straightforward cases to use v5.

The changes were made using a sed script for consistency. Some v4 usages
remain, as they require additional modifications to migrate to v5. These
will be addressed in a follow-up commit. This commit focuses solely on
mechanical changes to keep the review process simple.

Epic: none
Release note: none

Informs #137595

rafiss reviewed (approved):
> Reviewed 77 of 77 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139611)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @asg0451, @jeffswenson, @mgartner, and @xinhaoz)



<!-- Sent from Reviewable.io -->

rafiss reviewed (approved):
nice, glad so many of them were easy to upgrade!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/139611#-:-OHIdF7YDeY_ADgWNQbr:b-fzqqpb)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @asg0451, @jeffswenson, @mgartner, and @xinhaoz)



<!-- Sent from Reviewable.io -->

spilchen commented:
TFTR!

bors r+

spilchen commented:
bors r-

spilchen commented:
Whoops, I pushed another change to the wrong PR... Let me correct it.

spilchen commented:
bors r+
